pkg/scheduler/core/equivalence_cache.go

```diff
@@ -83,7 +83,9 @@ func (ec *EquivalenceCache) RunPredicate(
 	}
 	// Skip update if NodeInfo is stale.
 	if cache != nil && cache.IsUpToDate(nodeInfo) {
+		ec.mu.Lock()
 		ec.updateResult(pod.GetName(), nodeInfo.Node().GetName(), predicateKey, fit, reasons, equivClassInfo.hash)
+		ec.mu.Unlock()
```

Comment:
dito, make the function itself thread safe is better

Reply:
Actually, we need to hold the lock when we call cache.IsUpToDate. This is what fixed the race condition in https://github.com/kubernetes/kubernetes/pull/63040. I think the window for errors is extremely small with this setup though.

Reply:
ok, i saw that, i will keep the lock  outside of cache.IsUpToDate.